### PR TITLE
MULTISITE-19306: [TK] Check for unused modules

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: includes/drush/custom/custom.drush.inc [hunk 1]
```diff
@@ -20,6 +20,15 @@ function custom_drush_command() {
     'aliases' => ['fr-site', 'frsi'],
   ];
 
+  $items['check-unused-modules'] = [
```
Comment (jonhy81) [inline includes/drush/custom/custom.drush.inc:23]: Should we align and namespace all related drush commands introduced in 3.0.22?
```
- check-unused-modules
- platform-overridden-modules
- toolkit-contrib-authorised-security
- toolkit-contrib-update
```
@zarabatana , @verbruggenalex ?

File: includes/drush/custom/custom.drush.inc [hunk 2]
```diff
@@ -86,6 +95,36 @@ function drush_custom_features_revert_site() {
...
+  // If referenced make file does not exist, trow a warning.
+  if (!file_exists($make)) {
+    drush_log(dt('Make file @make does not exist. Showing all disabled modules in @path.', array('@make' => $make, '@path' => $path)), 'warning');
+  }
+  else {
+    $make_file = drupal_parse_info_file($make);
+  }
+
+  // Get all modules.
+  $modules = drupal_system_listing("/\\.module\$/", $path, 'name', 0);
```
Comment (jonhy81) [inline includes/drush/custom/custom.drush.inc:111]: There is a way to detect if site is available or not and log it to terminal if not?

**Review by jonhy81 — CHANGES_REQUESTED** [review]

File: includes/drush/custom/custom.drush.inc (hunk 1 above)
Comment (ghost) [reply includes/drush/custom/custom.drush.inc:23]: My proposal:
- toolkit-check-modules-unused
- toolkit-check-modules-platform
- toolkit-check-modules-authorized
- toolkit-check-modules-updated

Comment (jonhy81) [reply includes/drush/custom/custom.drush.inc:23]: I would say
```
toolkit-check-modules-unused
toolkit-check-modules-restricted
toolkit-check-modules-authorized
toolkit-check-modules-security-advisory
```

File: includes/drush/custom/custom.drush.inc (hunk 2 above)
Comment (ghost) [reply includes/drush/custom/custom.drush.inc:111]: I'll have a look at this to see if we can gracefully exit the command with a message.

Comment (jonhy81) [reply includes/drush/custom/custom.drush.inc:111]: Maybe ?
```
drush status bootstrap | grep -q Successful
```

File: includes/drush/custom/custom.drush.inc (hunk 1 above)
Comment (ghost) [reply includes/drush/custom/custom.drush.inc:23]: Agreed... I'll create a ticket to align the names.

File: includes/drush/custom/custom.drush.inc (hunk 2 above)
Comment (ghost) [reply includes/drush/custom/custom.drush.inc:111]: I checked what happens without a functional site:
```
    [error] Command check-unused-modules needs a higher bootstrap level to run - you will need to invoke drush from a more functional Drupal environment to run this command.
    [error] The drush command 'check-unused-modules' could not be executed.
```

This seems like an appropriate error message no?

**Pull request merged**